src/app/contexts/layout.tsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import buildContext from '~/components/jsx-helpers/build-context';
+import loadable from 'react-loadable';
+import LoadingPlaceholder from '~/components/loading-placeholder/loading-placeholder';
+
+// Webpack wasn't able to make the dynamic strings work in the Context value.
+type LayoutName = 'default' | 'landing';
+const loaders = {
+    'default': () => import('~/layouts/default/default'),
+    landing: () => import('~/layouts/landing/landing')
+};
+
+function useContextValue() {
+    const [layoutName, setLayoutName] = React.useState<LayoutName>('default');
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const [layoutData, setLayoutData] = React.useState<any>(undefined);
```

Comment:
i commented on this before but i forget, what was the issue with combining these two states into a 
```
React.useState<{ name: LayoutName; data: any;}>({name: 'default', data: undefined}); 
```
kind of thing?

Reply:
Need to be able to detect when a change of value isn't a real change, to prevent flicker or render cascades.
I was able to use `useReducer` instead of `useState` and have it work.

Reply:
ah right, because updating it to a new object using the same layout/params would be a different strict equality check, but with just a string and `undefined` it figures it out.